### PR TITLE
Preview.4 changes

### PR DESCRIPTION
**Pull request by Stevenic**
- The `LiveShareClient` class now requires a host.
- Removed any dependencies on `teams-js` package.
- Updated client to use `InsecureTokenProvider` for local tenants. Simplifies local testing.
- Incremented version and dependencies to `1.0.0-preview.4`
- Updated samples. They now use 2.5.0 of teams-js.

### CODE REVIEW

**Review by corinagum — APPROVED** [review]

**Review by pradeepananth — APPROVED** [review]
just minor doc update concern, looks good other than this

**Review by huntj88 — APPROVED** [review]
Code looks good, but saw small eslint and prettier errors

File: packages/live-share/src/LiveShareClient.ts [hunk 1]
```diff
@@ -62,48 +63,31 @@ export interface ILiveShareClientOptions {
...
 
     /**
      * Creates a new `TeamsFluidClient` instance.
+     * @param host Host for the current Live Share session.
      * @param options Optional. Configuration options for the client.
-     * @param host Optional. Host for the current Live Share session. If not specified the host
-     * will attempt to be automatically determined.
      */
-    constructor(options?: ILiveShareClientOptions, host?: ILiveShareHost) {
+    constructor(host: ILiveShareHost, options?: ILiveShareClientOptions) {
```
[the remark this answers is not in the document]
Comment (Stevenic) [reply packages/live-share/src/LiveShareClient.ts:79]: We need to update the docs separately but good point

File: packages/live-share/src/TestLiveShareHost.ts [hunk 2]
```diff
@@ -18,10 +18,22 @@ import {
 export class TestLiveShareHost implements ILiveShareHost {
     public static readonly LOCAL_MODE_TEST_TOKEN = `test-token`;
 
-    constructor(
-        private _getLocalTestContainerId?: () => string | undefined,
-        private _setLocalTestContainerId?: (containerId: string) => void
-    ) {}
+    /**
```
[the remark this answers is not in the document]
Comment (Stevenic) [reply packages/live-share/src/TestLiveShareHost.ts:21]: done

**Pull request merged**